ksqldb-parser/src/main/java/io/confluent/ksql/parser/properties/with/CreateSourceAsProperties.java

```diff
@@ -119,17 +120,26 @@ public SerdeFeatures getValueSerdeFeatures() {
   ) {
     final Builder<String, String> builder = ImmutableMap.builder();
     if (AvroFormat.NAME.equalsIgnoreCase(keyFormat)) {
-      // ensure that the schema name for the key is unique to the sink - this allows
-      // users to always generate valid, non-conflicting avro record definitions in
-      // generated Java classes (https://github.com/confluentinc/ksql/issues/6465)
-      builder.put(AvroFormat.FULL_SCHEMA_NAME, AvroFormat.getKeySchemaName(name));
+      final String schemaName = props.getString(CommonCreateConfigs.KEY_AVRO_SCHEMA_FULL_NAME);
+      if (schemaName != null) {
+        // Set if schema name is explicitly set
+        builder.put(AvroFormat.FULL_SCHEMA_NAME, schemaName);
+      } else {
+        // ensure that the schema name for the key is unique to the sink - this allows
+        // users to always generate valid, non-conflicting avro record definitions in
+        // generated Java classes (https://github.com/confluentinc/ksql/issues/6465)
+        builder.put(AvroFormat.FULL_SCHEMA_NAME, AvroFormat.getKeySchemaName(name));
```

Comment:
For my own education: why don't we have a similar schema name generation for the value below? And should we really introduce `KEY_AVRO_SCHEMA_FULL_NAME` ?

The linked tickets say:
> We could introduce KEY_AVRO_SCHEMA_FULL_NAME. Quickest solution... and lame.

Not sure if I understand the full context. Maybe @agavra knows?

Reply:
> why don't we have a similar schema name generation for the value below?

Because the purpose of this is to avoid clashing with `KsqlDataSourceSchema` which is the default name for value schema name? So if no name is provided, default value name will be `KsqlDataSourceSchema` and this will make sure key schema name will (may) be different from value schema name. So there's no need to generate value schema name? 

> We could introduce KEY_AVRO_SCHEMA_FULL_NAME. Quickest solution... and lame.

Maybe the original purpose is to get around clashing names instead of introduce ways to set key name for user? Not sure  exactly... @agavra can comment

Reply:
This is the suggestion in the original ticket that was implemented

> I propose we default the name of the record in the key schema to `<sourceNameInCamelCase>Key`, e.g. for a source named `BOB`, the record in the key schema would be called `BobKey`.

The reason Andy thought introducing `KEY_AVRO_SCHEMA_FULL_NAME` is "lame" is because it basically makes it a required property for anyone that's using AVRO for both keys/values (which isn't very SQLy), and is also trying to generate class names. It seems like better UX to just generate a good name (e.g. `BobKey`) from the beginning.

> For my own education: why don't we have a similar schema name generation for the value below?

@mjsax this is historical for backwards compatibility - I think it would make sense for us to generate the key/value both in the same way in the default path but we'd have to think through the implications to the users.

-----

All that being said, I'm not totally opposed to introducing `KEY_AVRO_SCHEMA_FULL_NAME` _if_ we have evidence that users aren't happy with our generated names. That gives them the control to potentially introduce also their own naming convention to the record names. But absent that user request, I'd err against introducing the added complexity.

Reply:
> it basically makes it a required property for anyone that's using AVRO for both keys/values

I don't fully understand this. It will be optional and the key schema name will be generated as `<sourceNameInCamelCase>Key` if not provided. Why it's required?

> we have evidence that users aren't happy with our generated names. That gives them the control to potentially introduce also their own naming convention to the record names. But absent that user request, I'd err against introducing the added complexity.

The context is that we want to control the key schema name and make it same as the avro schema's name we fetch from schema registry using provided schema id. Not sure if I'm a user :) Also I thought there's an existing `VALUE_AVRO_SCHEMA_FULL_NAME` and it doesn't hurt to provide `KEY_AVRO_SCHEMA_FULL_NAME` to add the flexibility as well. If there's downside of exposing `KEY_AVRO_SCHEMA_FULL_NAME` to user and there's cleaner ways to set the key schema name for avro format, it's also ok. What would be a cleaner way to set format property internally in general?

Reply:
> I don't fully understand this. It will be optional and the key schema name will be generated as <sourceNameInCamelCase>Key if not provided. Why it's required?

Sorry I wasn't totally clear, I meant that was the motivation for introducing the generated key name - now that we have that, it wouldn't be required. I just wanted to give context on the motivation.

> If there's downside of exposing KEY_AVRO_SCHEMA_FULL_NAME to user and there's cleaner ways to set the key schema name for avro format, it's also ok. 

I don't know of any other way, I don't think it's a serious downside other than just adding more bloat. I think we had imagined one day getting rid of `VALUE_AVRO_SCHEMA_FULL_NAME` and make it just the generated name, but I don't really have a strong opinion on whether that's the right thing to do or not.

tl;dr, I think it's fine to just keep your PR as is and introduce `KEY_AVRO_SCHEMA_FULL_NAME`.

Reply:
> I don't fully understand this. It will be optional and the key schema name will be generated as <sourceNameInCamelCase>Key if not provided. Why it's required?

I _think_ @agavra meant it would be required if we only have the property without any name generation.

> this is historical for backwards compatibility - I think it would make sense for us to generate the key/value both in the same way in the default path but we'd have to think through the implications to the users.

Thanks for the background. I guess the confusing part is, that we have a "value" property, that we also use to generate a key-name. It would be simpler, if the property would be name `AVRO_SCHEMA_FULL_NAME` (without the `VALUE_` prefix) and use `BobValue` and `BobKey` as names (to built upon the other example). But I understand why we ended up where we are.

> The context is that we want to control the key schema name and make it same as the avro schema's name we fetch from schema registry using provided schema id. Not sure if I'm a user :) 

That a new context and "you are not a user" (seems you are a program? :trollface:). For this new context it makes sense to add `KEY_AVRO_SCHEMA_FULL_NAME` now, but it did maybe not back in the days.

So I think we are good for now. Maybe a good follow up (out of scope for this PR) would be, to change existing behavior a little bit:
- If `KEY_AVRO_SCHEMA_FULL_NAME` is specified, use as-is (only for the key)
- If `VALUE_AVRO_SCHEMA_FULL_NAME` is specified, use as-is (only for the value -- _this is a change_)
- Add `AVRO_SCHEMA_FULL_NAME` and use for key and value with `XxxKey` and `XxxValue` names to avoid conflict

The tricky question is backward compatibility...

Reply:
lgtm

Reply:
> I don't know of any other way, I don't think it's a serious downside other than just adding more bloat. I think we had imagined one day getting rid of VALUE_AVRO_SCHEMA_FULL_NAME and make it just the generated name, but I don't really have a strong opinion on whether that's the right thing to do or not.

Even better. I guess we could try to use a combination of `cluster-id` and stream/table name as default: `MyClusterMyTableKey` and `MyClusterMyTableValue` and use `AVRO_SCHEMA_FULL_NAME` to let people overwrite it... Anyway. Not relevant for this PR.